File: backend/src/main/java/com/aplus/aplusmarket/documents/ChatMessage.java
package com.aplus.aplusmarket.documents;


import com.aplus.aplusmarket.dto.chat.request.ChatMessageDTO;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Setter
@Getter
@ToString
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "chatMessage")

/*
 * 2025.02.18 - 황수빈 : message를 기존 MySQL에서 mongo로 변경
 */

public class ChatMessage {

    @Id
    private String _id;
    private int userId;
    private int chatRoomId;
    private String content;
    private LocalDateTime createdAt;
    private LocalDateTime deletedAt;
    private Boolean isRead;

    // appointment 용 컬럼
    private String date;
    private String time;
    private String location;
    private String locationDescription;
    private int reminderBefore;



    public ChatMessageDTO toDTO(ChatMessage chatMessage){
        return ChatMessageDTO.builder()
                .chatRoomId(chatMessage.getChatRoomId())
                .messageId(chatMessage._id)
                .userId(chatMessage.getUserId())
                .content(chatMessage.getContent())
                .date(chatMessage.getDate())
                .time(chatMessage.getTime())
                .location(chatMessage.getLocation())
                .reminderBefore(chatMessage.getReminderBefore())
                .locationDescription(chatMessage.getLocationDescription())
                .createdAt(chatMessage.getCreatedAt())
                .isRead(chatMessage.getIsRead())
                .build();
    }
}

File: frontend/lib/data/model/chat/chat_message.dart
import 'package:flutter/material.dart';

class ChatMessage {
  final int chatRoomId;
  final String? messageId;
  final int userId;
  final String? content;
  final bool? isRead;
  final String? createdAt;
  final String? deletedAt;
  final String? date;
  final String? time;
  final String? location;
  final String? locationDescription;
  final int? reminderBefore;

  final bool? isFirst;

  ChatMessage(
      {required this.chatRoomId,
      required this.userId,
      this.messageId,
      this.content,
      this.isRead,
      this.createdAt,
      this.deletedAt,
      this.date,
      this.time,
      this.location,
      this.locationDescription,
      this.reminderBefore,
      this.isFirst = false});

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
        messageId: json['messageId'],
        userId: json['userId'],
        chatRoomId: json['chatRoomId'],
        content: json['content'] ?? json['date'],
        isRead: json['isRead'] ?? false,
        createdAt: json['createdAt'],
        deletedAt: json['deletedAt'] ?? '',
        date: json['date'],
        time: json['time'],
        location: json['location'],
        isFirst: json['isFirst']);
  }
  Map<String, dynamic> toJson() {
    return {
      'messageId': messageId,
      'chatRoomId': chatRoomId,
      'userId': userId,
      'date': date,
      'time': time,
      'location': location,
      'locationDescription': locationDescription,
      'isFirst': isFirst
    };
  }

  ChatMessage copyWith({
    String? messageId,
    int? chatRoomId,
    int? userId,
    String? content,
    String? date,
    String? time,
    String? location,
    String? locationDescription,
    int? reminderBefore,
    bool? isRead,
    String? createdAt,
  }) {
    return ChatMessage(
      messageId: messageId ?? this.messageId,
      chatRoomId: chatRoomId ?? this.chatRoomId,
      userId: userId ?? this.userId,
      content: content ?? this.content,
      date: date ?? this.date,
      time: time ?? this.time,
      location: location ?? this.location,
      locationDescription: locationDescription ?? this.locationDescription,
      reminderBefore: reminderBefore ?? this.reminderBefore,
      isRead: isRead ?? this.isRead,
      createdAt: createdAt ?? this.createdAt,
    );
  }

  @override
  String toString() {
    return 'ChatMessage{chatRoomId: $chatRoomId, messageId: $messageId, userId: $userId, content: $content, isRead: $isRead, createdAt: $createdAt, deletedAt: $deletedAt, date: $date, time: $time, location: $location, locationDescription: $locationDescription, reminderBefore: $reminderBefore}';
  }
}
